### PR TITLE
vsfproxy integration

### PR DESCRIPTION
**Pull request by siarizar**
Adding in integration for vsfproxy service

- Pass in the proxy endpoint using the instance variable
- The argument "verkada_department" is required, these are valid values:
  - Sales
  - Marketing
  - Engineering
  - Support

- Other added headers like client ID and username are optional and won't affect functionality if they aren't passed in
- If the proxy needs to be bypassed, pass in` instance = None`
- `is_proxy_accessible` function implemented to ensure that the proxy is accessible when initializing simple-salesforce. If the proxy is down, simple-salesforce will use the default salesforce endpoint instead

### CODE REVIEW

File: simple_salesforce/api.py [hunk 1]
```diff
@@ -251,13 +257,31 @@ def __init__(
                 'You must provide login information or an instance and token'
                 )
 
+        if instance is not None:
+            # Run a health check for the production vsfproxy, only assign 'self.sf_instance' if accessible
+            if instance == "vsfproxy.sops.vcamera.net/process_salesforce_request":
```
Comment (tejasbadadare) [inline simple_salesforce/api.py:262]: Pls extract into a global in case this endpoint changes in the future

**Review by tejasbadadare — COMMENTED** [review]
After merging can you tag the commit so we can refer to it easily in SalesforceSDK reqs?

**Review by warren-verkada — COMMENTED** [review]
Actually I'm thinking it might make more sense to keep this as a separate branch because: 

1. it's clearer why we've forked this by just reading the requirements file: `pip install git+ssh://github.com/verkada-com/simple-salesforce#vsfproxy-integration`
2. we can pull into `master` from upstream really easily and then rebase into the new branch

**Comment by tejasbadadare** [comment]
> Actually I'm thinking it might make more sense to keep this as a separate branch because:
> 
> 1. it's clearer why we've forked this by just reading the requirements file: `pip install git+ssh://github.com/verkada-com/simple-salesforce#vsfproxy-integration`
> 2. we can pull into `master` from upstream really easily and only then worry about merge conflicts when rebasing into the new branch

I'm thinking through if we ever need to add other features to our fork of simple-salesforce, in that case it would make more sense to have a canonical prod-deployable "main" branch rather than keep the "vsproxy-integration" branch up to date. We can also tag commits with human friendly names and reference those in pip requirements.

To your point about clarity on why this was forked -- lets update the README to have a section at the top that explains why our own fork is necessary since if anyone visits the repo homepage, currently its pretty unclear what the differences are between this and the upstream repo. 

cc @siarizar

**Comment by tejasbadadare** [comment]
> The argument "verkada_department" is required, these are valid values:
> 
> Sales
> Marketing
> Engineering
> Support

Just for my own knowledge - what happens if someone passes an unsupported department? will the proxy reject it since its a required header?

File: simple_salesforce/bulk2.py [hunk 2]
```diff
@@ -889,6 +886,16 @@ def _upload_file(
         workers = min(concurrency,
                       MAX_INGEST_JOB_PARALLELISM
                       )
+        
+        total_records = _count_csv(filename=csv_file,
```
[the remark this answers is not in the document]
Comment (siarizar) [reply simple_salesforce/bulk2.py:890]: Adjusted!

File: simple_salesforce/api.py (hunk 1 above)
Comment (siarizar) [reply simple_salesforce/api.py]: Done!

**Comment by siarizar** [comment]
> > The argument "verkada_department" is required, these are valid values:
> > Sales
> > Marketing
> > Engineering
> > Support
> 
> Just for my own knowledge - what happens if someone passes an unsupported department? will the proxy reject it since its a required header?

Yup the proxy will reject the request since it's a required header.

**Review by warren-verkada — APPROVED** [review]

**Pull request merged**